### PR TITLE
graph: sdpa: support dropout seed/offset/prob in fused sdpa

### PR DESCRIPTION
**Pull request by TaoLv**
For SDPA forward with dropout seed/offset/prob.

~~SDPA backward will be fixed later~~

Update: SDPA backward is also fixed.

### CODE REVIEW

**Comment by TaoLv** [comment]
Noticed correctness issue via benchdnn. Debugging...

```bash
# with fused sdpa kernel
$ _ONEDNN_GRAPH_SDPA_FORCE_PRIMITIVE=0 ./tests/benchdnn/benchdnn --graph --engine=gpu  --case=complex_fusion/mha/gqa-plain-training-fwd-w-dropout-bf16-f32.json
[COMPARE_STATS][DST]: trh=0 err_max_diff: 2.01562 err_max_rdiff:8.37618e+37 all_max_diff: 2.01562 all_max_rdiff:8.37618e+37
[COMPARE_STATS] Norm check is prohibited; error_to_total_ratio: 233469/262144; allowed_ratio: 256/262144;
Error: Function 'doit' at (/nfs/pdx/disks/hal9000/lvtao/oneDNN/tests/benchdnn/graph/graph.cpp:787) returned '1'
0:FAILED (errors:233469 total:262144) (3079 ms) __REPRO: --graph --engine=gpu --case=complex_fusion/mha/gqa-plain-training-fwd-w-dropout-bf16-f32.json
===========================================================
= Failed cases summary (--summary=no-failures to disable) =
===========================================================
0:FAILED (errors:233469 total:262144) (3079 ms) __REPRO: --graph --engine=gpu --case=complex_fusion/mha/gqa-plain-training-fwd-w-dropout-bf16-f32.json
============================
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total: 3.09s; create_pd: 0.09s (3%); create_prim: 0.96s (31%); fill: 0.00s (0%); execute: 0.01s (0%); compute_ref: 0.00s (0%); compare: 0.00s (0%);

# with primitive based kernel
$ _ONEDNN_GRAPH_SDPA_FORCE_PRIMITIVE=1 ./tests/benchdnn/benchdnn --graph --engine=gpu  --case=complex_fusion/mha/gqa-plain-training-fwd-w-dropout-
bf16-f32.json
0:PASSED (10122 ms) __REPRO: --graph --engine=gpu --case=complex_fusion/mha/gqa-plain-training-fwd-w-dropout-bf16-f32.json
tests:1 passed:1 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:0 listed:0
total: 10.12s; create_pd: 0.07s (1%); create_prim: 0.67s (7%); fill: 0.00s (0%); execute: 0.00s (0%); compute_ref: 0.00s (0%); compare: 0.00s (0%);
```

**Comment by TaoLv** [comment]
make test
set test_scope=NIGHTLY
disable benchdnn_all
enable benchdnn_graph

File: src/graph/interface/op_def.hpp [hunk 1]
```diff
@@ -2373,6 +2373,7 @@ DNNL_GRAPH_OP_SCHEMA(_sdpa, 1,
                 .set_attr(op_attr::is_invert_scale, false, attribute_kind::b,
                         false)
                 .set_attr(op_attr::is_training, false, attribute_kind::b)
+                .set_attr(op_attr::with_dropout, true, attribute_kind::b)
```
Comment (ElaineBao) [inline src/graph/interface/op_def.hpp]: the default value needs to be false, or setting it to false during fuse_sdpa pass when dropout does not exist.

Comment (TaoLv) [reply src/graph/interface/op_def.hpp]: Good catch. Fixed.

**Comment by TaoLv** [comment]
make test
set test_scope=NIGHTLY
disable benchdnn_all
enable benchdnn_graph

**Comment by h-sadia** [comment]
We are not enabling mask here? Also, we will need a backport to v3.12 branch as well.

**Comment by TaoLv** [comment]
> We are not enabling mask here? Also, we will need a backport to v3.12 branch as well.

Dropout mask output is not required for SDPA training in PyTorch.
Sure, I will backport these to rls-v3.12 once the correctness failures are addressed.

File: src/graph/backend/dnnl/executables/sdpa.cpp
Comment (h-sadia) [inline src/graph/backend/dnnl/executables/sdpa.cpp:58]: The attribute for sdpa has to be set in order for fused sdpa kernel

```suggestion
    if (with_dropout_) {
        dnnl::memory::desc dropout_mask_desc;
        attr.set_dropout(dropout_mask_desc, dnnl::memory::data_type::s64,
                /*use_offset*/ true, /*use_host_scalars*/ true);
    }
```

**Comment by TaoLv** [comment]
make test
disable benchdnn_all
enable benchdnn_graph

**Comment by TaoLv** [comment]
Adding commits from #4969 for validation. Will rebase the PR once #4969 is landed.

File: src/graph/backend/dnnl/executables/sdpa.cpp
Comment (TaoLv) [reply src/graph/backend/dnnl/executables/sdpa.cpp:58]: Even if we don't require the mask output, we still need to set this memory descriptor? What about the runtime mask output argument?

Comment (TaoLv) [reply src/graph/backend/dnnl/executables/sdpa.cpp:58]: I guess we don't need to pass the runtime mask output argument as here the `dropout_mask_desc` is empty, correct?

Comment (h-sadia) [reply src/graph/backend/dnnl/executables/sdpa.cpp:58]: yes, its an empty desc since we are not supporting mask. The function in attr().dropout_.has_default_values() checks for this; and bypasses the mask check if it is not enabled.

**Comment by TaoLv** [comment]
make test
disable benchdnn_all
enable benchdnn_graph

**Comment by TaoLv** [comment]
make test
disable benchdnn_all
enable benchdnn_graph
